### DISCUSSION TITLE
Question : Time synchronization for containers

### DISCUSSION THREAD

**Discussion by karlmutch** [k3s-io/k3s#4256] [Q&A]
I have been using several services within k3s deployments including rancher-desktop and kim.  Because I am running on a laptop the clock within various containers inside the cluster looses time while the machine is in a sleep state.

If I try to use ntpdate to synchronize things within containers I get issues with privileges and if I try to reset the clocks using hwclock also within containers I only ever see the delayed time that is affected by the host OS sleeping.

Some pods/containers I have benefit from having an ntpd deployment but the system pods etc never seem to recover the lost time.

How are we meant to be getting our clocks back into sync after sleeping etc ?

**Comment by brandond** [comment]
Containers don't have a separate system clock. Make sure your host time is set correctly and that will be shared with your containers. That said, Kubernetes is built for datacenter use and does not handle being run on a sleeping host well. You should probably restart K3s after waking at the very least.
